File: app/src/main/java/com/goranch/publicapis/ui/food/SearchRecipeView.java
package com.goranch.publicapis.ui.food;


import com.goranch.publicapis.api.model.food.recipe.Recipe;

import java.util.List;

public interface SearchRecipeView {

    void loadData(List<Recipe> recipeList);

    void showProgress();

    void hideProgress();

    void openDetailsFragment();

    void onItemClicked(Recipe mItem);
}

File: app/src/main/java/com/goranch/publicapis/ui/food/SearchRecipeView.kt
package com.goranch.publicapis.ui.food


import com.goranch.publicapis.api.model.food.recipe.Recipe

interface SearchRecipeView {

    fun loadData(recipeList: List<Recipe>)

    fun showProgress()

    fun hideProgress()

    fun openDetailsFragment()

    fun onItemClicked(mItem: Recipe)

    fun showHTTPError()
}
